frontend/public/components/utils/details-page.tsx

```diff
@@ -193,6 +194,43 @@ export const RuntimeClass: React.FC<RuntimeClassProps> = ({ obj, path }) => {
   );
 };
 
+export const UpstreamConfigDetailsItem: React.SFC<UpstreamConfigDetailsItemProps> = ({
+  resource,
+}) => {
+  const { t } = useTranslation();
+  // TODO this specific RBAC is duplicated across several modals and pages
+  const clusterVersionIsEditable =
+    useAccessReview({
+      group: ClusterVersionModel.apiGroup,
+      resource: ClusterVersionModel.plural,
+      verb: 'patch',
+      name: resource?.metadata?.name,
+    }) && window.SERVER_FLAGS.branding !== 'dedicated';
+  return (
+    <DetailsItem label={t('public~Upstream configuration')} obj={resource} path="spec.upstream">
+      <div>
+        <Button
+          type="button"
+          isInline
+          data-test-id="cluster-version-upstream-server-url"
+          onClick={(e) => {
+            e.preventDefault();
+            e.stopPropagation();
+            clusterVersionIsEditable && configureClusterUpstreamModal({ cv: resource });
+          }}
+          variant="link"
+          isDisabled={!clusterVersionIsEditable}
```

Comment:
Does this still render as a link or does PF render a disabled button with variant="link" as normal text?

If not, I think this is OK for now, but something we probably want to change in a follow up.

Reply:
@spadgett it renders a disabled button with variant="link". The text is grey versus blue:
<img width="1791" alt="Screen Shot 2021-09-01 at 1 05 04 PM" src="https://user-images.githubusercontent.com/21317056/131915555-5facc99d-03cc-4f32-b4e6-f19e70e5112e.png">